### ISSUE TITLE
Example settings [documentation request]

### ISSUE THREAD

**Issue by bmakuh** [cspotcode/vscode-mocha-latte#9]
First off, this is a neat idea for an extension, but so far I've been unable to actually get it to work because I can't seem to configure it correctly. So as a documentation request, I'd love to see an example workspace `settings.json` to get this to work, especially when using Babel as a mocha compiler.

**Comment by bmakuh** [comment]
To be clear, it's probably that I'm just dense and need help 😄

**Comment by betososa** [comment]
@bmakuh if you go to `File > Preferences > Workspace Setting` and browse to the `Mocha Options` you will see there are only 5 settings

- mocha.options: i think you can pass here `--reporter spec`

- mocha.files.glob: where you have your `describe`s functions

- mocha.files.ignore: who would like to skip the `node_modules` folder

- mocha.env: env variables to test with

- mocha.node_options: idk

Hope it can help you

**Comment by bmakuh** [comment]
Thanks @beturs! These are the settings I wrote based on the readme, but they don't work:
```json
    "mocha.options": {
        "opts": "./test/mocha.opts",
        "compilers": "js:babel-core/register",
        "timeout": 15000,
        "recursive": true,
        "R": "dot"
    },
    "mocha.files.glob": "app/**/*.test.js",
    "mocha.files.ignore": [
        "**/.git/**/*",
        "**/node_modules/**/*"
    ]
```

**Comment by betososa** [comment]
i found this issue in the parent repo https://github.com/compulim/vscode-mocha/issues/4 
i think its related
:(

**Comment by cspotcode** [comment]
How about creating a Github wiki page or section for examples?  That will
allow us to copy-paste any examples from this thread into a more permanent
home, and we can link to it from the readme.

**Comment by batjko** [comment]
Any news on this?
I'd settle for a single example snippet in the README. Specifically how to add babel, which is probably what most people will need.

**Comment by cspotcode** [comment]
If anyone writes such an example on the wiki, I'll link to it from the
README.

The example you're asking for (using Mocha with Babel) sounds like a Mocha
tutorial, not specifically a vscode-mocha-latte tutorial.
 vscode-mocha-latte, after all, is merely a few IDE commands that
ultimately execute Mocha under the hood.  Mocha is where all the magic
happens.

I think it would be easiest for someone to find a Mocha tutorial that
already exists.  We can either link directly to that tutorial or someone
can adapt it into a vscode-mocha-latte configuration and post it on the
wiki.

**Comment by batjko** [comment]
@cspotcode Not really. Using Babel with Mocha is well documented and no problem on the command line (i.e.`mocha --compilers js:babel-register`).
So all I need to know is where to put this same information in the settings for this vscode extension.
As @bmakuh pointed out above, it's not exactly self-evident.

**Comment by cspotcode** [comment]
Does this snippet from Mocha's documentation answer the question?

From https://mochajs.org/#compilers:
> **About Babel**
> 
> If your ES6 modules have extension `.js`, you can `npm install --save-dev babel-register` and use `mocha --require babel-register`; `--compilers` is only necessary if you need to specify a file extension.

vscode-mocha-latte has a `mocha.requires` option, so I think this'll work.

If someone puts the relevant information into a wiki page, I'll update the README to link to it.  This allows the community to keep the wiki updated without a need to merge PRs into master.  vscode-mocha-latte's wiki is here: https://github.com/cspotcode/vscode-mocha-latte/wiki

EDIT: it's `mocha.requires`, not `mocha.require`.  Fixed the typo.

**Comment by batjko** [comment]
@cspotcode Thanks for the tip! I've tried that and it seems to work. 
I say "seems" because I ran into another issue after that (I use jsdom to render react components into a fake DOM, and was unable to add that configuration, but that's likely a different issue).

So I gave up on this extension for now, since it wouldn't given me anything more than what I get when running `npm test` and setting up the extension is too much of a hassle, so I'll stick to just the terminal for the time being.

**Comment by bmakuh** [comment]
@batjko yeah, pretty much the same for me. Requiring just `babel-register` is a 💯 tip, but now when I try to require `test/support/dom.js` (where I configure jsdom) I get an error that it can't be found.

**Comment by cspotcode** [comment]
Happy to help, I'm glad it's working.  If you run into other problems that
are specifically *bugs in vscode-mocha-latte* then please file them as
separate issues with an explanation for how to reproduce the bug.  Fixing
bugs is very important to me.  Documentation is something I think the
community can handle best and something that I don't have the time to do
myself.

On Feb 17, 2017 10:11 AM, "Ben" <notifications@github.com> wrote:

@batjko <https://github.com/batjko> yeah, pretty much the same for me.
Requiring just babel-register is a 💯 tip, but now when I try to require
test/support/dom.js (where I configure jsdom) I get an error that it can't
be found.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/cspotcode/vscode-mocha-latte/issues/9#issuecomment-280675602>,
or mute the thread
